optaplanner-quickstarts/quarkus-school-timetabling/src/test/java/org/acme/schooltimetabling/solver/TimeTableConstraintProviderTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.schooltimetabling.solver;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+import org.acme.schooltimetabling.domain.Lesson;
+import org.acme.schooltimetabling.domain.Room;
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.domain.Timeslot;
+import org.junit.jupiter.api.Test;
+import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+
+class TimeTableConstraintProviderTest {
```

Comment:
Not public?

Reply:
Since JUnit 5, it needn't be. And I would argue it shouldn't be, as it doesn't export nothing to any other class, nor does it need to be visible.

Most of Optaplanner code makes the test class public. However, the constraint provider in facility maintenance quickstart doesn't. So, faced with a choice, I picked the rule that was closer and made more sense.